Docs: Add section for troubleshooting "Definition for rule x was not found" when doing Local Linking

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

After performing the local linking steps, I am getting: `Definition for rule '@typescript-eslint/no-unused-vars' was not found` when running `npx eslint .`

It feels like the page might lack information on what the minimum requirements for a repository made to test a rule are, but if the steps on the page are in fact the only ones, they don't seem to work for me. I ran:
```
~/typescript-eslint (main)
$ for package in ./packages/*; do
    cd $package
    npm link
    cd ../..
done
```
which succeeded after adding a bunch of packages, then:
```
~/playground (main)
$ npm link @typescript-eslint/eslint-plugin @typescript-eslint/parser

added 2 packages, changed 1 package, and audited 386 packages in 4m
```
and then:
```
 ~/playground (main)
$ npx eslint .

  1:1  error  Definition for rule '@typescript-eslint/no-unused-vars' was not found  @typescript-eslint/no-unused-vars
```

My .eslintrc.json looks like this:
```
{
	"rules": {
		"@typescript-eslint/no-unused-vars": "error"
	}
}
```
and I'm pretty confident that's a real rule

### Affected URL(s)

https://typescript-eslint.io/contributing/local-development/local-linking

--- a/docs/contributing/local-development/Local_Linking.mdx
+++ b/docs/contributing/local-development/Local_Linking.mdx
@@ -10,6 +10,10 @@ The general strategy to do so is:
 2. [Repository linking](#repository-linking): Use your package manager's link command to reference those global symlinks in the local downstream repository.
 3. [Trying rules](#trying-rules): Test your local rules and plugins by enabling them in the local downstream repository.
 
+:::tip
+Before linking a repository, make sure it's gone through the steps in [Getting Started](../../getting-started/Quickstart.mdx) to be able to use typescript-eslint rules.
+:::
+
 ## Global Linking
 
 To make all `@typescript-eslint/*` packages available globally, run this command from your `typescript-eslint` repository root:
@@ -29,9 +33,15 @@ The command to put instead of that `#` comment, depends on the local downstream 
 - [Yarn v1 / classic](https://classic.yarnpkg.com/lang/en/docs/cli/link/ 'Yarn v1 / classic docs'): `yarn link`
 - [Yarn v2 / v3 / berry](https://yarnpkg.com/cli/link 'Yarn v2 / v3 / berry docs'): _skip this step altogether_
 
+Additionally, if you haven't yet built the typescript-eslint repository, do so now:
+
+```shell
+yarn build
+```
+
 ## Repository Linking
 
-Now that the `@typescript-eslint/*` packages are available locally, you can link to them in the local downstream repository.
+Now that the `@typescript-eslint/*` packages are built and available locally, you can link to them in the local downstream repository.
 Run that repository's package manager's link command for any `@typescript-eslint/*` packages that repository depends on:
 
 - npm: `npm link @typescript-eslint/eslint-plugin @typescript-eslint/parser`
@@ -84,6 +94,15 @@ In this case, you can manually install any missing packages in the local downstr
 ```shell
 yarn add ts-api-utils -D
 ```
+
+### Rules Not Found (`Definition for rule ... was not found`)
+
+Packages need to be built.
+First try re-running `yarn build` from the typescript-eslint repository root.
+
+If you're adding new rules, you'll need to also modify [`packages/eslint-plugin/src/rules/index.ts`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/index.ts) to export the rule object, then re-build.
+
+If you're modifying preset configurations, you'll also need to run `yarn generate-configs`, then re-build.
 
 ### Yarn Link Failures (`Cannot link ... conflicts with parent dependency`)
 

